src/mapview.jl

```diff
@@ -0,0 +1,3 @@
+import .SplitApplyCombine: MappedArray
+
+Base.setindex!(ma::MappedArray, val, ix) = parent(ma)[ix] = set(parent(ma)[ix], ma.f, val)
```

Comment:
The feature is very neat, but this is [type piracy](https://docs.julialang.org/en/v1/manual/style-guide/#Avoid-type-piracy). One possibility would be to create a zero dependencies package `AccessorsBase` that just contains something like
```julia
function set end
```
and ask @andyferris if we can add the `Base.setindex!` overload at SplitApplyCombine.

Reply:
That would also be piracy: all these functions, at least https://github.com/JuliaObjects/Accessors.jl/blob/master/src/functionlenses.jl

Reply:
Maybe, "pirating" a method that isn't defined, and won't feasibly be defined in the original library is fine? As I understand, the main concern is that piracy can change the behavior of unrelated code - here it's hard to imagine such a case. SAC.jl cannot really define setindex for an arbitrary mappedarray.

Reply:
> That would also be piracy: all these methods, at least https://github.com/JuliaObjects/Accessors.jl/blob/master/src/functionlenses.jl

That is fine by the 

> Sometimes, coupled packages may engage in type piracy to separate features from definitions

clause in the manual. If we overloaded `UnsupectingPackage.set` that would be a problem.

> Maybe, "pirating" a method that isn't defined, and won't feasibly be defined in the original library is fine? 

Fair enough. @andyferris do you think we can do this overload here?

Reply:
> Maybe, "pirating" a method that isn't defined, and won't feasibly be defined in the original library is fine? 

What can happen is that some other people think the same and provide overloads for `Base.setindex!(::MappedArray...)` Then a third package might have `Accessors` and the other package as transitive dependencies and get weird bugs.

Reply:
This PR doesn't introduce any new dependency, `Requires` is already used in `Accessors`.

Reply:
Ok I just saw the growinh Project.toml. Thats what people will look at to check dependencies.

Mostly everyone is moving away from Requires.jl because its a hack with serious ttfx overheads - it cant precompile.

Reply:
Accessors.jl getting heavy is a good point @rafaqz and getting rid of Requires.jl is also on my wishlist. Currently, we only need it to avoid depending on StaticArrays.jl . 
Do you have a suggestion where stuff like this PR could live?

Reply:
Well we really conditional dependencies: https://github.com/JuliaLang/Pkg.jl/issues/1285

What ArrayInterface.jl is doing and we're currently doing with GeoInterface.jl is making subpackages in the same repository, names as the combination of packages. Like ArrayInterfaceOffsets.jl or GeoInterfaceTables.jl.

Then other packages and users can choose their dependencies. Its pretty clunky but the best we have currently. 

(also sorry I meant to put my initial comment in the main thread I commented from a phone)

Reply:
I'm also curious how the StaticArrays support in Accessors would look without Requires. Should this https://github.com/JuliaObjects/Accessors.jl/blob/master/src/staticarrays.jl file be put into a separate package that depends on both? The major issue then is that users wouldn't install it and will think that Accessors just doesn't support SVectors.